### ISSUE TITLE
PURCHASEHISTORY isn't working

### ISSUE THREAD

**Issue by influence412** [chr233/ASFEnhance#274]
**Describe the bug / Bug 描述**
Describe what the bug is
简单描述一下是什么 bug

**Error Log / 错误日志**
Please paste the error log of ASFEnhance, for example
请在此粘贴 ASFEnhance 的错误日志, 示例如下

```txt
ASFenhance 遇到错误, 日志如下
==========================================
 - 原始消息: TEST
 - Access: Owner
 - ASF 版本: 5.4.0.3
 - 插件版本: 1.7.5.0
==========================================
{
  "EULA": true,
  "Statistic": true,
  "DevFeature": false
}
==========================================
 - 错误类型: System.Exception
 - 错误消息: Exception of type 'System.Exception' was thrown.
   at ASFEnhance.ASFEnhance.ResponseCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)
   at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)
```

Fix this command, please
![image](https://github.com/chr233/ASFEnhance/assets/154758173/dbc52b91-e386-4b70-b246-afee157577c1)

**Comment by chr233** [comment]
![image](https://github.com/chr233/ASFEnhance/assets/13798576/d7b10b4c-72b2-4a0a-a0ac-faac8505e4a7)

mine works, what's region your bot are?

**Comment by influence412** [comment]
> ![image](https://private-user-images.githubusercontent.com/13798576/344243897-d7b10b4c-72b2-4a0a-a0ac-faac8505e4a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1OTYxNTQsIm5iZiI6MTcxOTU5NTg1NCwicGF0aCI6Ii8xMzc5ODU3Ni8zNDQyNDM4OTctZDdiMTBiNGMtNzJiMi00YTBhLWEwYWMtZmFhYzg1MDVlNGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI4VDE3MzA1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ0MjNhNjcyYzAyNWRkMWU1YmQ0ZGJiZGNlZGZkNmIzZmE3ODFiZWIwNDZjODA1YWRmODE1YjIwMGY3OGVhMzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pBORxEVZWbIz72gPvB74cNnHddfzqgC4lXZcV1tWUwQ)
> 
> mine works, what's region your bot are?

RU

**Comment by influence412** [comment]
> ![image](https://private-user-images.githubusercontent.com/13798576/344243897-d7b10b4c-72b2-4a0a-a0ac-faac8505e4a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1OTYzMjEsIm5iZiI6MTcxOTU5NjAyMSwicGF0aCI6Ii8xMzc5ODU3Ni8zNDQyNDM4OTctZDdiMTBiNGMtNzJiMi00YTBhLWEwYWMtZmFhYzg1MDVlNGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI4VDE3MzM0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1Yzg5MzI2M2MwNDE0YThiMGU5NjRiZjhlOGJiZDZhNWRlYzE2MGVjYzM2NjRmYTU4Y2M1MTY0M2ZhOWQxOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ky-D3R-xYbGLVdmpM03B64JGx5Oji8LKea4G66Byn1g)
> 
> mine works, what's region your bot are?

![image](https://github.com/chr233/ASFEnhance/assets/154758173/4d4ab012-9fef-4280-be96-b3a47857a68e)
USA isnt working too

**Comment by chr233** [comment]
> > ![image](https://private-user-images.githubusercontent.com/13798576/344243897-d7b10b4c-72b2-4a0a-a0ac-faac8505e4a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1OTYzMjEsIm5iZiI6MTcxOTU5NjAyMSwicGF0aCI6Ii8xMzc5ODU3Ni8zNDQyNDM4OTctZDdiMTBiNGMtNzJiMi00YTBhLWEwYWMtZmFhYzg1MDVlNGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI4VDE3MzM0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1Yzg5MzI2M2MwNDE0YThiMGU5NjRiZjhlOGJiZDZhNWRlYzE2MGVjYzM2NjRmYTU4Y2M1MTY0M2ZhOWQxOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ky-D3R-xYbGLVdmpM03B64JGx5Oji8LKea4G66Byn1g)
> > mine works, what's region your bot are?
> 
> ![image](https://private-user-images.githubusercontent.com/154758173/344261719-4d4ab012-9fef-4280-be96-b3a47857a68e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk2MjU4NTksIm5iZiI6MTcxOTYyNTU1OSwicGF0aCI6Ii8xNTQ3NTgxNzMvMzQ0MjYxNzE5LTRkNGFiMDEyLTlmZWYtNDI4MC1iZTk2LWIzYTQ3ODU3YTY4ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyOVQwMTQ1NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjczMzQ3MmE3MzliOTk0Y2M2OTRmYTkzNjYyMjE3ZDJjNDI5ZWY4Zjk1NDIwNzgwNTgxOTRkZjhjNzg4OGI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hV_TCrY1PPajh_Iy43x6ifCbCyq7p1ZgtMVzrH3RtKE) USA isnt working too

![image](https://github.com/chr233/ASFEnhance/assets/13798576/855de4b7-e3c7-415a-ab3f-2d92136e36f6)
mine works

**Comment by chr233** [comment]
> > ![image](https://private-user-images.githubusercontent.com/13798576/344243897-d7b10b4c-72b2-4a0a-a0ac-faac8505e4a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1OTYzMjEsIm5iZiI6MTcxOTU5NjAyMSwicGF0aCI6Ii8xMzc5ODU3Ni8zNDQyNDM4OTctZDdiMTBiNGMtNzJiMi00YTBhLWEwYWMtZmFhYzg1MDVlNGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI4VDE3MzM0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1Yzg5MzI2M2MwNDE0YThiMGU5NjRiZjhlOGJiZDZhNWRlYzE2MGVjYzM2NjRmYTU4Y2M1MTY0M2ZhOWQxOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ky-D3R-xYbGLVdmpM03B64JGx5Oji8LKea4G66Byn1g)
> > mine works, what's region your bot are?
> 
> ![image](https://private-user-images.githubusercontent.com/154758173/344261719-4d4ab012-9fef-4280-be96-b3a47857a68e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk2MjU4NTksIm5iZiI6MTcxOTYyNTU1OSwicGF0aCI6Ii8xNTQ3NTgxNzMvMzQ0MjYxNzE5LTRkNGFiMDEyLTlmZWYtNDI4MC1iZTk2LWIzYTQ3ODU3YTY4ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyOVQwMTQ1NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjczMzQ3MmE3MzliOTk0Y2M2OTRmYTkzNjYyMjE3ZDJjNDI5ZWY4Zjk1NDIwNzgwNTgxOTRkZjhjNzg4OGI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hV_TCrY1PPajh_Iy43x6ifCbCyq7p1ZgtMVzrH3RtKE) USA isnt working too

RU not work confirmed

**Comment by chr233** [comment]
fix wip

**Comment by chr233** [comment]
![image](https://github.com/chr233/ASFEnhance/assets/13798576/dde003ed-748b-4f5d-884f-96181b232874)

seems regex pattern not match `pуб.`

**Comment by chr233** [comment]
fixed in 2.1.10.4
